controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -72,23 +74,66 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	clientSet, err := kubeclientset.NewForConfig(cachedCluster.RestConfig)
 	if err != nil {
 		reqLogger.Error(err, "cannot create ClientSet for the ToolchainCluster")
+		if err := r.updateStatus(ctx, toolchainCluster, clusterNotReadyCondition()); err != nil {
+			reqLogger.Error(err, "unable to update cluster status of ToolchainCluster")
+		}
 		return reconcile.Result{}, err
 	}
-	healthChecker := &HealthChecker{
-		localClusterClient:     r.Client,
-		remoteClusterClient:    cachedCluster.Client,
-		remoteClusterClientset: clientSet,
-		logger:                 reqLogger,
-	}
+
+	// execute healthcheck
+	healthcheckResult := r.getClusterHealthCondition(ctx, clientSet)
+
 	// update the status of the individual cluster.
-	if err := healthChecker.updateIndividualClusterStatus(ctx, toolchainCluster); err != nil {
+	if err := r.updateStatus(ctx, toolchainCluster, healthcheckResult...); err != nil {
 		reqLogger.Error(err, "unable to update cluster status of ToolchainCluster")
 		return reconcile.Result{}, err
 	}
-
 	return reconcile.Result{RequeueAfter: r.RequeAfter}, nil
 }
 
+func (r *Reconciler) updateStatus(ctx context.Context, toolchainCluster *toolchainv1alpha1.ToolchainCluster, currentconditions ...toolchainv1alpha1.Condition) error {
+	toolchainCluster.Status.Conditions = condition.AddOrUpdateStatusConditionsWithLastUpdatedTimestamp(toolchainCluster.Status.Conditions, currentconditions...)
+	if err := r.Client.Status().Update(ctx, toolchainCluster); err != nil {
+		return fmt.Errorf("failed to update the status of cluster - %s "+err.Error(), toolchainCluster.Name)
+	}
+	return nil
+}
+
+func (r *Reconciler) getClusterHealthCondition(ctx context.Context, remoteClusterClientset *kubeclientset.Clientset) []v1alpha1.Condition {
+
+	healthcheck, errhealth := r.getClusterHealth(ctx, remoteClusterClientset)
+	if errhealth != nil || !healthcheck {
+		return []v1alpha1.Condition{clusterNotReadyCondition()}
```

Comment:
As I mentioned in the previous review and also during the sync call, let's use a proper `Reason` based on the result of the health check.
* if error is not nil, then `ClusterNotReachable` and also pass the error as the message to the condition
* if health check is false, then `ToolchainClusterClusterNotReadyReason`

We wanted to reuse the condition only, not the reason.

Reply:
updated it now